.github/CONVENTIONS.md

```diff
@@ -0,0 +1,132 @@
+
+#### Variable names
+Keep variable names as short as possible and as long as needed to convey their meaning. E.g. `int vehicleArrayCount` could be turned into `int vehiclesCount` or just `int nVehicles`.
+
+### Type prefixes
+All member variables follow the pattern `<m=Member,s=Static>_<one character type><PascalCaseName>` as per the AR guidelines. 
+This project diverges slighty from their enforcement. Only primitive types *need* to be prefixed, anything that is not prefixed is assume a complex type.
+In case of collections (which are also complex types) `Array` and `Set` share the same prefix **a** to avoid confusion with string primtives and their API is similar.
+If the prefix adds no relevant information for complex types it can be omitted e.g. `m_pUserEntity` -> `m_UserEntity` as per AR guidelines.
+AR uses some type alias to give primitives more meaning such as `ResourceName`. You can optionally annotate those e.g. `m_rPrefab`. The fallback is the root primitive they are based on.
+
+| Type     | Boolean     | Integer     | Enum        | Float       | String      | Typename     | Vector      | Array       | Set         | Map         | Class       |
+|----------|-------------|-------------|-------------|-------------|-------------|--------------|-------------|-------------|-------------|-------------|-------------|
+| Optional | No          | No          | No          | No          | No          | No           | No          | **Yes**     | **Yes**     | **Yes**     | **Yes**     |
+| Example  | m_**b**Name | m_**i**Name | m_**e**Name | m_**f**Name | m_**s**Name | m_**t**Name  | m_**v**Name | m_**a**Name | m_**a**Name | m_**m**Name | m_**p**Name |
+
+### ComponentEditorProps and EntityEditorProps
+The attributes use to decorate `EntityClass` and `ComponentClass` classes follow the feature pattern described in the directory structure. Examples:
+- `[EntityEditorProps(category: "EveronLife/Feature/Housing")]`
+- `[ComponentEditorProps(category: "EveronLife/Feature/Farming/Mining")]`
+
+### Modded class tagged names
+When adding class members or functions throug a `modded` class include the TAG e.g. `EL` in the name to avoid conflicts. E.g. `m_iELModdedCount` or `void EL_Update()`.
+
+### Single line statements
+Single line statements that can be used on `if/else`, `for`, `foreach`, `while` are tolerated, but should be kept to a minimum to make the code readable and easy to debug/change.
+Good use cases in include:
+- `if (condition) return;`
+- `if (condition) continue;`
+- `if (instance) instance.Init();`
+
+If the code becomes too complex, the line too long or the likely hood of wanting to debug something in there is reasonably high, put it into a code block with curly braces.
+
+### Use of the auto keyword
+The keyword `auto` is only allowed if the right hand side expression is disclosing the expected compile time type.  
```

Comment:
I dont really understand this.
I thought if you know the variable type you should not use auto and you know its "EL_SomeVeryLongComponentName" so why use auto?

Reply:
Because the line gets way too long and it is a lot to type. If the type is guaranteed by the right hand side and can be read without the need to go to the source and look it up, it is redundant information. Auto is meant for exactly this situation, same as `var` in c#.